lib/engine/game/g_1849/game.rb

```diff
@@ -771,12 +824,137 @@ def bank_sort(corporations)
         end
 
         def player_value(player)
-          player.value - @player_debts[player]
+          player.value - @player_debts[player] - player.shares_by_corporation.sum do |corp, _|
+            player.num_shares_of(corp) * corp.loans.size * 100
+          end
         end
 
         def par_prices
           @stock_market.share_prices_with_types(@available_par_groups)
         end
+
+        # code below is for variant rules
+
+        def acquiring_station_tokens?
+          @acquiring_station_tokens ||= @optional_rules&.include?(:acquiring_station_tokens)
+        end
+
+        def electric_dreams?
+          @electric_dreams ||= @optional_rules&.include?(:electric_dreams)
+        end
+
+        def bonds?
+          @bonds ||= @optional_rules&.include?(:bonds)
+        end
+
+        def reduced_4p_corps?
+          @reduced_4p_corps ||= @optional_rules&.include?(:reduced_4p_corps)
+        end
+
+        def event_buy_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:buy_tokens][1]} --"
+        end
+
+        def event_bonds!
+          @log << "-- Event: #{EVENTS_TEXT[:bonds][1]} --"
+        end
+
+        def event_e_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:e_tokens][1]} --"
+        end
+
+        # code below is for Bonds variant
+        NUM_LOANS = 6
+        MAXIMUM_LOANS = 1
+        LOAN_VALUE = 500
+        INTEREST_RATE = 50
+
+        def init_loans
+          Array.new(num_loans) { |id| Loan.new(id, loan_value) }
+        end
+
+        def num_loans
+          bonds? ? NUM_LOANS : 0
+        end
+
+        def maximum_loans(_entity)
+          MAXIMUM_LOANS
```

Comment:
self.class::MAXIUM_LOANS

Applies to other constants too.

Reply:
Done, except for this one, which doesn't seem to allow that format without crashing my fixture games:

```
        def num_loans
          bonds? ? NUM_LOANS : 0
        end
```

is that expected?

Reply:
commit `add self.class to all definitions`